[ADD] inventory: package use and type (#7734)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/advanced_operations_warehouse/package.rst

========
Packages
========

.. |SO| replace:: :abbr:`SO (Sales Order)`

A *package* is a physical container holding one or more products. Packages can also be used to store
items in bulk.

Packages are commonly used for the following purposes:

#. :ref:`Relocating products from stock <inventory/warehouses_storage/package-use>` to packing or
   outbound areas for delivery.

> #. :ref:`Shipping to customers <inventory/warehouses_storage/package-type>`: configure package types
>    to align with shipping carriers' size and weight requirements, streamlining the packing process,
>    and ensuring compliance with carrier shipping specifications.

Reviewer: ```suggestion
#. :ref:`Shipping to customers <inventory/warehouses_storage/package-type>`, in which package types
   are configured to align with shipping carriers' size and weight requirements, streamlining the
   packing process, and ensuring compliance with carrier shipping specifications.
```

#. Storing items in bulk.

*Package use* is a field on the package form in Odoo that is only visible by enabling the *Batch

> Transfers* feature in settings of the *Inventory* application.

Reviewer: ```suggestion
Transfers* feature in the settings of the *Inventory* application.
```

   workflow involving storable products.

.. seealso::
   :doc:`Ship one order in multiple packages
   <../../shipping_receiving/advanced_operations_shipping/multipack>`

.. _inventory/warehouses_storage/enable-package:

Configuration
=============

To use packages, first go to :menuselection:`Inventory app --> Configuration --> Settings`. Under
the :guilabel:`Operations` heading, activate the :guilabel:`Packages` feature. Then, click
:guilabel:`Save`.

.. image:: package/enable-pack.png
   :align: center
   :alt: Activate the *Packages* setting in Inventory > Configuration > Settings.

.. _inventory/warehouses_storage/package-use:

Package use
===========

.. note::
   The following section is **only** necessary when packages are intended to be used for cluster
   pickings.

.. seealso::
   - :doc:`Picking method: batch <batch_transfers>`
   - :doc:`Picking method: cluster <cluster_picking>`
   - :doc:`Picking method: wave <wave_transfers>`

First, navigate to :menuselection:`Inventory app --> Configuration --> Settings`, and activate the
:guilabel:`Batch Transfers` feature, located in the :guilabel:`Operations` section. Doing so makes

> the *Package Use* field become visible on a package form.

Reviewer: ```suggestion
the *Package Use* field appear on a package form.
```

   :align: center
   :alt: Display package form to create a cluster pack.

Add to batch
------------

.. important::
   The following example uses packages in a :doc:`three-step delivery
   <../../shipping_receiving/daily_operations/delivery_three_steps>` workflow.

   This workflow requires the :guilabel:`Packages`, :guilabel:`Storage Locations`, and
   :guilabel:`Multi-Step Routes` features to be enabled in :menuselection:`Inventory app -->
   Configuration --> Settings`.

   On the warehouse configuration page (:menuselection:`Inventory app --> Configuration -->
   Warehouses`), the :guilabel:`Outgoing Shipments` option **must** be set to :guilabel:`Pack goods,
   send goods in output and then deliver (3 steps)`.

After confirming a sales order, selecting the :guilabel:`Delivery` smart button displays the three
tasks to complete the delivery.

.. image:: package/three-step.png
   :align: center
   :alt: Clicking "Delivery" smart button displays the pick, pack, ship steps.

To create a batch transfer, navigate to :menuselection:`Inventory app --> Operations --> Internal`
and select the checkboxes next to the pick operations that are to be added to the batch.

> After the pick operations are selected, navigate to the :guilabel:`⚙️ (gear) Actions` button, and

Reviewer: ```suggestion
After the pick operations are selected, navigate to the :guilabel:`Actions ⚙️ (gear)` icon button, and
```

creating the batch.

.. seealso::
   :ref:`More info about the fields in the 'Add to batch' window
   <inventory/warehouses_storage/add-batch-transfers>`

Pack items
----------

After creating the batch, open the batch transfer (:menuselection:`Inventory app --> Operations -->
Batch Transfers`), and select a reusable package (created in the :ref:`previous step
<inventory/warehouses_storage/package-use>`) in the :guilabel:`Destination Package` field to place
items in the package.

.. example::
   The reusable package, `CLUSTER-PACK-1`, is assigned to the :guilabel:`Destination Package` field
   of all items in the batch transfer, `BATCH/00003`.

  .. image:: package/assign-package.png
     :align: center
     :alt: Assign pickings to a package through the "Destination Package" field.

Alternatively, click the :guilabel:`Put in Pack` button to create a new package, and place all the
items in the batch in that newly-created package.

.. note::
   Ensure the new package's :guilabel:`Package Use` field is set to :guilabel:`Reusable` by clicking
   the package's :guilabel:`➡️ (right arrow)` icon to open the package form.

   .. image:: package/pack-internal-link.png
      :align: center

>       :alt: Show internal link

Author: Need a period at the end of the alt tag here, but in the next update PR: https://github.com/odoo/documentation/pull/7797

This image gets removed, so I wouldn't worry about this (;

   outbound areas for delivery.
#. :ref:`Shipping to customers <inventory/warehouses_storage/package-type>`: configure package types
   to align with shipping carriers' size and weight requirements, streamlining the packing process
   and ensuring compliance with carrier shipping specifications.
#. Storing items in bulk.

Refer to the table below for essential package configurations based on different use cases. Skip to
the corresponding sections by selecting the blue link to configure the necessary features for the
intended package usage.

.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * -
     - Package use
     - Package type
   * - Batch transfer
     - Reusable
     - Optional
   * - Deliver to customer
     - Disposable
     - Recommended
   * - Store in bulk
     - Disposable
     - Optional

*Package use* is a field on the package form in Odoo that is only visible by enabling the
:guilabel:`Batch Transfers` feature in settings. The default package type is :guilabel:`Disposable`,
so this feature is only required when packages are intended to be used for batch transfers and other

> internal warehouse moves.

Author: Thanks for this reminder about using italics in this case! I'll keep this in mind for this and future docs :D

opens the package form, which contains the following fields:

- :guilabel:`Package Reference`: (required) name of the package
- :guilabel:`Pack Date`: the date that the package was created
- :guilabel:`Package Use`: choose :guilabel:`Reusable` for packages used for moving products within
  the warehouse; :guilabel:`Disposable` for packages used to ship products to customers.
- :guilabel:`Package Type`: used for :ref:`configuring shipping boxes to ship to the customer
  <inventory/warehouses_storage/package-type>`.

  .. note::
     :guilabel:`Package Type` is unnecessary for configuring batch transfer packages.

- :guilabel:`Shipping Weight`: used to input the weight of the package after measuring it on a
  scale.
- :guilabel:`Company`: specify a company to make the package available **only** at the selected
  company. Leave the field blank if the package is available at all companies.

.. image:: package/package.png
   :align: center
   :alt: Display package form to create a cluster pack.

Add to batch
------------

.. important::
   The following example uses packages in a :doc:`three-step delivery
   <../../shipping_receiving/daily_operations/delivery_three_steps>` workflow.

>    This workflow requires the :guilabel:`Packages`, :guilabel:`Storage Locations`, and
>    :guilabel:`Multi-Step Routes` features to be enabled in :menuselection:`Inventory app -->
>    Configuration --> Settings`. On the warehouse configuration page (:menuselection:`Inventory app
>    --> Configuration --> Warehouses`), the :guilabel:`Outgoing Shipments` option must be set to
>    :guilabel:`Pack goods, send goods in output and then deliver (3 steps)`.

Reviewer: ```suggestion
   This workflow requires the :guilabel:`Packages`, :guilabel:`Storage Locations`, and
   :guilabel:`Multi-Step Routes` features to be enabled in :menuselection:`Inventory app -->
   Configuration --> Settings`.
   
   On the warehouse configuration page (:menuselection:`Inventory app --> Configuration -->
   Warehouses`), the :guilabel:`Outgoing Shipments` option **must** be set to :guilabel:`Pack goods,
   send goods in output and then deliver (3 steps)`.
```

After confirming a sales order, selecting the :guilabel:`Delivery` smart button displays the three
tasks to complete the delivery.

.. image:: package/three-step.png
   :align: center
   :alt: Clicking "Delivery" smart button displays the pick, pack, ship steps.

To create a batch transfer, navigate to :menuselection:`Inventory app --> Operations --> Internal`

> and select the checkboxes next to the pick operations that are to be added to the batch.

Reviewer: I'm not seeing an 'Internal' option under the Operations menu in the Inventory app (while using a 16.0 runbot)
https://drive.google.com/file/d/1nKVQIrugAi8kfLnAejCxhZrdIApG34Dk/view?usp=sharing

Reviewer: Are you intending them to navigate to the Internal Transfers page, from the Internal Transfers card from the Inventory Overview page? If so, some clearer directions would be super helpful.

> After the pick operations are selected, navigate to the :guilabel:`⚙️ Actions` button and select
> :guilabel:`Add to batch` from the list of drop-down options.

Reviewer: ```suggestion
After the pick operations are selected, navigate to the :guilabel:`⚙️ (gear) Actions` button, and
select :guilabel:`Add to batch` from the list of drop-down menu options.
```

.. image:: package/create-batch.png
   :align: center
   :alt: Add pickings to batch.

Doing so opens an :guilabel:`Add to batch` pop-up window. Click :guilabel:`Confirm` to finish

> creating the batch.

Reviewer: is there any benefit to briefly explaining the fields present on the 'Add to batch' pop-up? Maybe even adding a screenshot could be beneficial.

Author: I was thinking, since batches aren't the focus on this packages doc, I didn't take a screenshot here or go into much details about the fields. Instead, I linked a related doc in the seealso block right below this paragraph. 

Do you think that gives enough details? I wanted to get your opinion as a non-Inventory expert. If you think the lack of explanation about batches disrupts your understanding of packages and their use case here, I can go into further detail about batches (:

.. seealso::
   :ref:`More info about the fields in the 'Add to batch' window
   <inventory/warehouses_storage/add-batch-transfers>`

Pack items
----------

> After creating the batch, open the batch transfer and select a reusable package (created in the
> :ref:`previous step <inventory/warehouses_storage/package-use>`) in the :guilabel:`Destination
> Package` field to place items in the package.

Reviewer: ```suggestion
After creating the batch, open the batch transfer, and select a reusable package (created in the
:ref:`previous step <inventory/warehouses_storage/package-use>`) in the :guilabel:`Destination
Package` field to place items in the package.
```

.. example::
   The reusable package, `CLUSTER-PACK-1`, is assigned to the :guilabel:`Destination Package` field
   of all items in the batch transfer, `BATCH/00003`.

  .. image:: package/assign-package.png
     :align: center
     :alt: Assign pickings to a package through the "Destination Package" field.

> Alternatively, click the :guilabel:`Put in Pack` button to create a new package and place all the

Reviewer: ```suggestion
Alternatively, click the :guilabel:`Put in Pack` button to create a new package, and place all the
```

> items in the batch in that newly created package.

Reviewer: ```suggestion
items in the batch in that newly-created package.
```

.. note::
   Ensure the new package's :guilabel:`Package Use` field is set to :guilabel:`Reusable` by clicking
   the package's :guilabel:`➡️ (right arrow)` icon to open the package form.

   .. image:: package/pack-internal-link.png
      :align: center
      :alt: Show internal link

.. example::

   Clicking :guilabel:`Put in Pack` creates a new package, `PACK0000002`, and assigns all items to
   it in the :guilabel:`Destination Package` field.

  .. image:: package/put-in-pack.png
     :align: center
     :alt: Image of the "Put in Pack" button being clicked.

.. _inventory/warehouses_storage/package-type:

Package type
============

Create package types at :menuselection:`Inventory app --> Configuration --> Package Types` to set

> custom dimensions and weight limits. This feature is mainly used to set templates and ensure

Reviewer: ```suggestion
custom dimensions and weight limits. This feature is mainly used to set templates, and ensure
```

packages comply with the shipping requirements of third-party carriers.

.. seealso::
   :doc:`Shipping carriers <../../shipping_receiving/setup_configuration/third_party_shipper>`

On the :guilabel:`Package Types` list, clicking :guilabel:`New` opens a blank package type form. The
fields of the form are as follows:

- :guilabel:`Package Type` (required): define the package type's name.

> - :guilabel:`Size`: define the dimensions of the package in millimeters (mm). The fields from left
>   to right define the :guilabel:`Length`, :guilabel:`Width`, and :guilabel:`Height`.

Reviewer: ```suggestion
- :guilabel:`Size`: define the dimensions of the package in millimeters (mm). The fields, from left
  to right, define the :guilabel:`Length`, :guilabel:`Width`, and :guilabel:`Height`.
```

tasks to complete the delivery.

.. image:: package/three-step.png
   :align: center
   :alt: Clicking "Delivery" smart button displays the pick, pack, ship steps.

To create a batch transfer, navigate to :menuselection:`Inventory app --> Operations --> Internal`
and select the check boxes next to the pick operations that are to be added to the batch.

After the pick operations are selected, navigate to the :guilabel:`⚙️ Actions` button and select
:guilabel:`Add to batch` from the list of drop-down options.

.. image:: package/create-batch.png
   :align: center
   :alt: Add pickings to batch.

Doing so opens an :guilabel:`Add to batch` pop-up window, and click :guilabel:`Confirm` to finish
creating the batch.

.. seealso::
   :ref:`More info about the fields in the 'Add to batch' window
   <inventory/warehouses_storage/add-batch-transfers>`

Pack items
----------

After creating the batch, open the batch transfer and select a reusable package (created in the
:ref:`previous step <inventory/warehouses_storage/package-use>`) in the :guilabel:`Destination
Package` field to place items in the package.

.. example::

>    The reusable package, `CLUSTER-PACK-1` is assigned to the :guilabel:`Destination Package` field

Reviewer 2: ```suggestion
   The reusable package, `CLUSTER-PACK-1`, is assigned to the :guilabel:`Destination Package` field
```

   of all items in the batch transfer, `BATCH/00003`.

  .. image:: package/assign-package.png
     :align: center
     :alt: Assign pickings to a package through the "Destination Package" field.

Alternatively, click the :guilabel:`Put in Pack` button to create a new package and place all the
items in the batch in that newly created package.

.. note::
   Ensure the new package's :guilabel:`Package Use` field is set to :guilabel:`Reusable` by clicking
   the package's :guilabel:`➡️ (right arrow)` icon to open the package form.

   .. image:: package/pack-internal-link.png
      :align: center
      :alt: Show internal link

.. example::

   Clicking :guilabel:`Put in Pack` creates a new package, `PACK0000002`, and assigns all items to
   it in the :guilabel:`Destination Package` field.

  .. image:: package/put-in-pack.png
     :align: center
     :alt: Image of the "Put in Pack" button being clicked.

.. _inventory/warehouses_storage/package-type:

Package type
============

Create package types at :menuselection:`Inventory app --> Configuration --> Package Types` to set

> custom dimensions and weight limits. This feature is mainly to set templates and ensure packages to
> comply with the shipping requirements of third-party carriers.

Reviewer 2: ```suggestion
custom dimensions and weight limits. This feature is mainly used to set templates and ensure packages
comply with the shipping requirements of third-party carriers.
```

.. seealso::
   :doc:`Shipping carriers <../../shipping_receiving/setup_configuration/third_party_shipper>`

On the :guilabel:`Package Types` list, clicking :guilabel:`New` opens a blank package type form. The

> following fields of the form are as follows:

Reviewer 2: ```suggestion
fields of the form are as follows:
```

- :guilabel:`Package Type` (required): define the package type's name.
- :guilabel:`Size`: define the dimensions of the package in millimeters (mm). The fields from left
  to right define the :guilabel:`Length`, :guilabel:`Width`, and :guilabel:`Height`.
- :guilabel:`Weight`: record the theoretical weight of the package type. This value is a placeholder
  for when the weight of the contained products is not available.
- :guilabel:`Max Weight`: maximum shipping weight allowed in the package.
- :guilabel:`Barcode`: define a barcode to identify the package type from a scan.
- :guilabel:`Company`: specify a company to make the package type available **only** at the selected
  company. Leave the field blank if it is available at all companies.

> - :guilabel:`Carrier`: specifies the intended shipping carrier for this package type.

Reviewer 2: ```suggestion
- :guilabel:`Carrier`: specify the intended shipping carrier for this package type.
```
